src/notifier/mqtt_notifier.py

```diff
@@ -0,0 +1,121 @@
+# std
+import json
+import logging
+from typing import List
+
+# project
+from . import Notifier, Event
+
+
+class MqttNotifier(Notifier):
+    def __init__(self, title_prefix: str, config: dict):
+        logging.info("Initializing MQTT notifier.")
+        super().__init__(title_prefix, config)
+
+        self._username = None
+        self._password = None
+
+        try:
+
+            credentials = config["credentials"]
+            self._set_config(config)
+            self._set_credentials(credentials)
+
+        except KeyError as key:
+            logging.error(f"Invalid config.yaml. Missing key: {key}")
+
+        self._init_mqtt()
+
+    def _set_config(self, config: dict):
+
+        self._topic = config["topic"]
+        self._qos: int = config["qos"] if "qos" in config else 0
+        self._retain: bool = config["retain"] if "retain" in config else False
+
+    def _set_credentials(self, credentials: dict):
+
+        self._host = credentials["host"]
+        self._port: int = credentials["port"]
+
+        if "username" in credentials and credentials["username"] != "":
+            self._username = credentials["username"]
+        if "password" in credentials and credentials["password"] != "":
+            self._password = credentials["password"]
+
+    def _init_mqtt(self) -> bool:
+
+        try:
+            client = __import__("paho.mqtt.client", globals(), locals(), ["Client"], 0)
+            self._client = client.Client()
+
+        except ImportError:
+            logging.error(
+                "The Paho MQTT module was not found, please refer to INTEGRATIONS.md for help on resolving"
+                "this issue. The integration will not function correctly until this issue is resolved."
+            )
+
+            return False
+
+        self._client.on_connect = self._on_connect
+        self._client.on_disconnect = self._on_disconnect
+
+        if self._username and self._password:
+            self._client.username_pw_set(self._username, self._password)
+
```

Comment:
Not sure if that is preferred behavior. It is possible (though probably unlikely) to target a public MQTT server without authentication.

Reply:
I think it's more likely going to be a misconfigured config file rather than intentional non-authenticated server. For the latter case the person would probably know what they're doing and can ignore that warning log message about missing credentials :-)